manifests/certificate/x509.pp

```diff
@@ -159,7 +159,7 @@
   $_csr = pick($csr, "${_csr_dir}/${name}.csr")
   $_key = pick($key, "${_key_dir}/${name}.key")
 
-  if !empty($altnames+$extkeyusage) {
+  if $altnames or $extkeyusage {
```

Comment:
What's about `$req_ext = !$altnames.empty or !$extkeyusage.empty` instead of that if-else-spaghetti

Reply:
well, you have to detect if we actually have values in extkexusage and altnames in order to set the req_ext. You cant set it always as that would error out when generating a cert, when both arrays are empty (and not set in the config, assuming its fixed). I am working on memory here though, when i last worked on it. I could be wrong

Reply:
First: My suggesting is essentially to use
```
$boolean_variable = (expression)
```
over something like
```
if (expression) {
  $boolean_variable = true
} else {
  $boolean_variable = false
}
```

Second, the `v3_req` section of the config depends on a lot more parameters. The `$req_ext` is not passed to the template at all, it is referenced only at a single point, to pass it to `x509_cert`. Thus, you can drop this variable and replace the expression into the `x509_cert` to be passed to its `req_ext` parameter.